packages/@react-stately/numberfield/src/useNumberFieldState.ts

```diff
@@ -54,92 +48,37 @@ export function useNumberFieldState(
     value,
     defaultValue,
     onChange,
-    inputValueFormatter,
-    numberParser,
-    locale,
-    currentNumeralSystem,
-    setCurrentNumeralSystem
+    locale
   } = props;
 
-  let intlOptions = useMemo(() => inputValueFormatter.resolvedOptions(), [inputValueFormatter]);
+  let [numberValue, setNumberValue] = useControlledState<number>(value, isNaN(defaultValue) ? NaN : defaultValue, onChange);
+  let [inputValue, setInputValue] = useState(() => isNaN(numberValue) ? '' : getNumberFormatter(locale, formatOptions).format(numberValue));
+
+  let numberingSystem = useMemo(() => getNumberingSystem(inputValue), [inputValue]);
+  let formatter = useMemo(() => getNumberFormatter(locale, {...formatOptions, numberingSystem}), [locale, formatOptions, numberingSystem]);
```

Comment:
getNumberFormatter already has a cache, do we want or need to cache it again?

Reply:
Yeah but we have to compute the cache key still... 🤷